mono/mini/aot-runtime.c

```diff
@@ -4152,8 +4152,11 @@ load_method (MonoDomain *domain, MonoAotModule *amodule, MonoImage *image, MonoM
 			}
 			return NULL;
 		}
+
```

Comment:
Doing this check here, after we actually accessed the method entry above looks fishy. Also, shouldn't the calling code make sure to pass a correct method_index.

Reply:
good point, I moved the bound check before the access.

passing an invalid `method_index` is okay I think (see function comment).